### PR TITLE
Impl `ErgoDataBridge`

### PR DESCRIPTION
**Pull request by kettlebell**

### CODE REVIEW

**Review by oskin1 — COMMENTED** [review]
We don't need to track all on-chain events, but rather only those, which satisfy a certain predicate (e.g. Contain boxes with certain ErgoTreeTemplate).

**Comment by kettlebell** [comment]
> We don't need to track all on-chain events, but rather only those, which satisfy a certain predicate (e.g. Contain boxes with certain ErgoTreeTemplate).

Right. Since `DataBridge` returns the receivers, the consumers (protocol handlers) can filter as they wish. Now perhaps different PHs filter in different ways and we're multicasting the TX stream from a central location, so we need to send every TX. Are my assumptions correct here?

**Comment by oskin1** [comment]
> > We don't need to track all on-chain events, but rather only those, which satisfy a certain predicate (e.g. Contain boxes with certain ErgoTreeTemplate).
> 
> Right. Since `DataBridge` returns the receivers, the consumers (protocol handlers) can filter as they wish. Now perhaps different PHs filter in different ways and we're multicasting the TX stream from a central location, so we need to send every TX. Are my assumptions correct here?

Not exactly. Let's start from "Within one node: one bridge = one consumer = one predicate (one set of predicates)". Cannot imagine multiple consumers for one chain.

File: Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ members = [
     "spectrum-crypto",
     "spectrum-vrf",
     "spectrum-move",
+    "spectrum-data-bridge",
```
Comment (oskin1) [inline Cargo.toml]: Let's keep connectors for each chain in separate crates
```suggestion
    "spectrum-ergo-connector",
```

**Review by oskin1 — APPROVED** [review]

**Pull request merged**